torch/_dynamo/variables/dicts.py

```diff
@@ -44,12 +44,14 @@ def is_hashable(x):
                 variables.ConstantVariable,
                 variables.EnumVariable,
                 variables.user_defined.UserDefinedClassVariable,
+                variables.UserFunctionVariable,
                 variables.misc.SkipFilesVariable,
                 variables.misc.NumpyVariable,
                 variables.NNModuleVariable,
                 variables.MethodWrapperVariable,
                 variables.TorchInGraphFunctionVariable,
                 variables.TypingVariable,
+                variables.FunctoolsPartialVariable,
```

Comment:
These need tests :)

Reply:
As it says in the OP, this is teseted by adding `fullgraph=True` to the test from a previous PR. That being said, I'll add a one-off tests for both of these types.

Reply:
Oh sorry I missed that.